endpoint.go

```diff
@@ -222,6 +228,15 @@ func (ep *endpoint) isAnonymous() bool {
 	return ep.anonymous
 }
 
+func (ep *endpoint) needDNSServer() bool {
+	if ep.getNetwork().Type() == "null" || ep.getNetwork().Type() == "host" {
+		return false
+	}
```

Comment:
Except the legacy default bridge network related things isn't it better to do any other network specific checks in libnetwork ?